device/config/functions.go

```diff
@@ -0,0 +1,227 @@
+
+	if _, ok := storage[GetConstLiteral(Wifi)+"_name"]; !ok {
+		return nil
+	}
+
+	ssh, ok := storage["ssh"].(ssh_helper.Util)
+	if !ok {
+		return errors.New("Cannot get ssh config")
+	}
+
+	fp := help.AddPathSuffix("unix", constants.MountDir, constants.ISAAX_CONF_DIR, "wpa_supplicant", constants.WPAsupplicant)
+	data := fmt.Sprintf(constants.WPAconf, storage[GetConstLiteral(Wifi)+"_name"], storage[GetConstLiteral(Wifi)+"_pass"])
+
+	_, eut, err := ssh.Run(fmt.Sprintf(`echo "%s" > %s`, data, fp))
+	if err != nil {
+		return errors.New(err.Error() + ":" + eut)
+	}
+
+	return nil
+}
+
+// ConfigInterface is a dialog asking to setup user Interfaces for the static ip functionality
+func ConfigInterface(storage map[string]interface{}) error {
+	device := []string{"eth0", "wlan0"}
+	i := Interfaces{
+		Address: "192.168.0.254",
+		Netmask: "255.255.255.0",
+		Gateway: "192.168.0.1",
+		DNS:     "192.168.0.1",
+	}
+
+	if dialogs.YesNoDialog("Would you like to assign static IP address for your device?") {
+		fmt.Println("[+] Available network interface: ")
+		num := dialogs.SelectOneDialog("Please select a network interface:", device)
+		fmt.Println("[+] ********NOTE: ADJUST THESE VALUES ACCORDING TO YOUR LOCAL NETWORK CONFIGURATION********")
+
+		for {
+			fmt.Printf("[+] Current values are:\n \t[+] Address:%s\n\t[+] Gateway:%s\n\t[+] Netmask:%s\n\t[+] DNS:%s\n",
+				i.Address, i.Gateway, i.Netmask, i.DNS)
+
+			if dialogs.YesNoDialog("Change values?") {
+				SetInterfaces(&i)
+
+				switch device[num] {
+				case "eth0":
+					storage[GetConstLiteral(Interface)] = fmt.Sprintf(constants.InterfaceETH, i.Address, i.Netmask, i.Gateway, i.DNS)
+					fmt.Println("[+]  Ethernet interface configuration was updated")
+				case "wlan0":
+					storage[GetConstLiteral(Interface)] = fmt.Sprintf(constants.InterfaceWLAN, i.Address, i.Netmask, i.Gateway, i.DNS)
+					fmt.Println("[+]  wifi interface configuration was updated")
+				}
+			} else {
+				break
+			}
+		}
+	}
+
+	return nil
+}
+
+// SaveInterface is a default method to save user Interfaces into the image
+func SaveInterface(storage map[string]interface{}) error {
+
+	if _, ok := storage[GetConstLiteral(Interface)]; !ok {
+		return nil
+	}
+
+	ssh, ok := storage["ssh"].(ssh_helper.Util)
+	if !ok {
+		return errors.New("Cannot get ssh config")
+	}
+
+	fp := help.AddPathSuffix("unix", constants.MountDir, constants.ISAAX_CONF_DIR, "network", constants.InterfacesF)
+
+	_, eut, err := ssh.Run(fmt.Sprintf(`echo "%s" > %s`, storage[GetConstLiteral(Interface)], fp))
+	if err != nil {
+		return errors.New(err.Error() + ":" + eut)
+	}
+
+	return nil
+}
+
+// ConfigSecondaryDns is a dialog asking to set 8.8.8.8 DNS
+func ConfigSecondaryDns(storage map[string]interface{}) error {
+	if dialogs.YesNoDialog("Add Google DNS as a secondary NameServer") {
+		storage[GetConstLiteral(Dns)] = true
+		return nil
+	}
+
+	return nil
+}
+
+// SaveInterface is a default method to set 8.8.8.8 as a secondary DNS
```

Comment:
__[golint]__
* [golint] comment on exported function SaveSecondaryDns should be of the form "SaveSecondaryDns ..."

<!-- Comment by [SideCI](https://sideci.com) -->